### PR TITLE
fix recursion error when handling unsupported types

### PR DESCRIPTION
**Pull request by mpangrazzi**
Should fix https://github.com/deepset-ai/hayhooks/issues/43.

Seems like some nested generic types are causing `handle_unsupported_types` to return `RecursionError: maximum recursion depth exceeded`.

### CODE REVIEW

**Comment by mpangrazzi** [comment]
I've iterated on this:

- rewrite of `handle_unsupported_types` (simpler + add tests)
- add a logger
- add IT tests of deploying sample pipelines using API, check status and `/docs` route

This should solve both #43 and #47.

File: tests/test_files/chat_with_website.yaml [hunk 1]
```diff
@@ -1,50 +0,0 @@
-components:
```
Comment (anakin87) [inline tests/test_files/chat_with_website.yaml:1]: why are we removing this file?

**Review by anakin87 — COMMENTED** [review]
I'm OK with the changes.

Can we make the tests run in the CI?
We can do this in another PR. Ping me if you need help.

File: tests/test_files/chat_with_website.yaml (hunk 1 above)
Comment (mpangrazzi) [reply tests/test_files/chat_with_website.yaml:1]: @anakin87 as discussed, I've just added and tested the working example of `chat_with_website` pipeline

**Review by anakin87 — APPROVED** [review]
Looks reasonable.

Feel free to merge, but let's work soon on running tests in the CI.

**Pull request merged**